### DISCUSSION TITLE
Run Talos ControlPlane As Pods

### DISCUSSION THREAD

**Discussion by infinitydon** [siderolabs/talos#8251] [Ideas]
Hi,

Not sure if this has been considered already to run Talos controlplanes as Pods within an existing k8s cluster, this will be an alternative to using dedicated VMs/baremetals for running the controlplane, examples of existing projects that does is k0smotron, kubermatic etc.

This will ultimately require using the operator pattern.

Thanks

**Comment by smira** [comment]
You can already do that by running Talos as containers, see https://www.talos.dev/v1.6/talos-guides/install/local-platforms/docker/#running-talos-in-docker-manually for the guidance.

There's no published operator for it, but certainly doable.

There are some security concerns with that, but there is nothing which blocks you from doing it.

**Comment by infinitydon** [comment]
Understood. Thanks for the response
